### ISSUE TITLE
[$1000] Blue line appears around avatar while user close the preview avatar by pressing Esc key.

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#17419]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Login with any account.
2. Try to preview user avatar, click on image to zoom, click again to make it becomes normal size.
3. Hide the preview modal by pressing Esc key.
4. Notice that there's a blue line around avatar before it completely dismissed.

## Expected Result
There's no blue line appears while hiding the modal by pressing Esc key.
## Actual Result
There's a blue line around avatar before it completely dismissed.

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** v1.3.0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/43996225/231887283-581b0a29-a2f4-4ba9-9875-b644832fa54a.mov

https://user-images.githubusercontent.com/43996225/231887295-e5fadd4f-1a5e-4e9e-8b09-37d79c554bdf.mp4

**Expensify/Expensify Issue URL:**
**Issue reported by:**  @hungvu193 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1681377270755139

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01104442472fdfdf76</li>
        <li>Upwork Job ID: 1648159313658621952</li>
        <li>Last Price Increase: 2023-05-09</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
Triggered auto assignment to @anmurali (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by MelvinBot** [comment]
@anmurali Uh oh! This issue is overdue by 2 days. Don't forget to update your issues!

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~01104442472fdfdf76

**Comment by MelvinBot** [comment]
Current assignee @anmurali is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @Santhosh-Sellavel (`External`)

**Comment by anmurali** [comment]
Able to reproduce, but you have to watch real closely when you hit escape!

**Comment by MelvinBot** [comment]
Triggered auto assignment to @Beamanator (`External`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by bernhardoj** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Blue focus outline is shown when we press the image and then press ESC.

### What is the root cause of that problem?
This is a normal behavior on web where a focusable element will show a focus outline when we press it first (to give it focus) and press any key (not necessarily esc). The focusable element in this case is the image wrapped by Pressable.
https://github.com/Expensify/App/blob/88644166f51c9f7ea8211df0b9afee1ca4784700/src/components/ImageView/index.js#L275-L287

### What changes do you think we should make in order to solve the problem?

I don't think this is an issue because it's the normal behavior of web. But if we insist to not show the outline, we can prevent the image to get the focus both by keyboard and mouse by adding these 2 props to the Pressable.
```
onMouseDown={e => e.preventDefault()}
focusable={false}
```

**Comment by kmorales13** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
There's a blue line around avatar before it completely dismissed.

### What is the root cause of that problem?
By default, when a `modal` is closed, the last focused element inside the `modal` will be focused before the `modal` is closed. This is the expected behavior for accessibility reasons, as it helps users to maintain their place in the page and understand what has happened.

In this scenario, the last focused element is the `image`, however this also applies to the `backdrop` itself, if we open a modal and close with `Esc` we will also see the blue outline for the `backdrop` as it is the first focused element when opening the `modal`.

### What changes do you think we should make in order to solve the problem?
If we want to prevent this behavior, we could `blur` the last active element just before closing the modal as such:
```
onModalWillHide={() => {
  if (document.activeElement) {
    document.activeElement.blur()
  }
}}```

**Comment by MelvinBot** [comment]
📣 @kmorales13! 📣

Hey, it seems we don’t have your contributor details yet! You'll only have to do this once, and this is how we'll hire you on Upwork.
Please follow these steps:

1. Get the email address used to login to your Expensify account. If you don't already have an Expensify account, create one [here](https://new.expensify.com/). If you have multiple accounts (e.g. one for testing), please use your main account email.
2. Get the link to your Upwork profile. It's necessary because we only pay via Upwork. You can access it by logging in, and then clicking on your name. It'll look like [this](https://www.upwork.com/freelancers/~01ca8ad8ce36b9eceb). If you don't already have an account, sign up for one [here](https://www.upwork.com/nx/signup/?dest=home).
3. Copy the format below and paste it in a comment on this issue. Replace the placeholder text with your actual details.

![Screen Shot 2022-11-16 at 4 42 54 PM](https://user-images.githubusercontent.com/30609178/231263719-a8a3b7d3-36a0-4d7f-8ad4-cebe1f560fa4.png)

Format:
```
Contributor details
Your Expensify account email: <REPLACE EMAIL HERE>
Upwork Profile Link: <REPLACE LINK HERE>
```

**Comment by situchan** [comment]
This is duplicated with https://github.com/Expensify/App/issues/10599. We agreed to do nothing and GH closed.

**Comment by kmorales13** [comment]
Contributor details
Your Expensify account email: kmoralesdev@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~0192c832b8e3913386

**Comment by MelvinBot** [comment]
✅ Contributor details stored successfully. Thank you for contributing to Expensify!

**Comment by Santhosh-Sellavel** [comment]
> This is duplicated with #10599. We agreed to do nothing and GH closed.

cc: @Beamanator @anmurali thoughts?

**Comment by Beamanator** [comment]
@Santhosh-Sellavel @situchan it looks like the other issue (https://github.com/Expensify/App/issues/10599) related to the avatar image on the Settings page, but this issue deals with the avatar image on the Details page - so I don't think they're necessarily duplicates. In this issue, the image on the details page gets the border, though it's not an Input so it doesn't really make sense to me for that image to get the blue border

**Comment by Beamanator** [comment]
@Santhosh-Sellavel it looks like we're still waiting on you to review proposals 👍

**Comment by Santhosh-Sellavel** [comment]
@Beamanator Kinda lost what happened here, I remember leaving a comment here.

All looks like a workaround to me, let's wait for new proposals thanks!

**Comment by MelvinBot** [comment]
📣 It's been a week! Do we have any satisfactory proposals yet? Do we need to adjust the bounty for this issue? 💸

**Comment by Santhosh-Sellavel** [comment]
@huzaifa-99 

Why are you suggesting using TouchableWithoutFeedback here?

**Comment by Santhosh-Sellavel** [comment]
That does look like a workaround to me.

**Comment by bernhardoj** [comment]
Just tested and we can zoom in (currently throws an error)/out the image by focusing the image and press enter. I think it makes sense to have the blue outline focus for the image?

Btw, here is the error when we try to zoom in with enter key:
<img width="588" alt="image" src="https://user-images.githubusercontent.com/50919443/234737395-e89f41d0-7446-4b7d-89fb-cccc196cc1df.png">

**Comment by jliexpensify** [comment]
@Santhosh-Sellavel hello! Just wondering if this issue might be linked/have a similar resolution? https://github.com/Expensify/App/issues/18052

They both deal with a blue line around the border. Thanks!

**Comment by MelvinBot** [comment]
@Beamanator @anmurali @Santhosh-Sellavel this issue was created 2 weeks ago. Are we close to approving a proposal? If not, what's blocking us from getting this issue assigned? Don't hesitate to create a thread in #expensify-open-source to align faster in real time. Thanks!

**Comment by Beamanator** [comment]
@anmurali  Looks like we should double the price of this one, what do you think @Santhosh-Sellavel?

**Comment by bernhardoj** [comment]
@Beamanator in case you missed my [comment](https://github.com/Expensify/App/issues/17419#issuecomment-1524396931), I think it's fine to have the blue border because we can interact (zoom in/out) with the avatar by keyboard. However, the zoom in throws an error and I think we should fix that instead here. 

The cause of the error because it is trying to read the position from the pointer event, but because we use keyboard (enter), the pointer event is undefined. One way to overcome this is to fallback the position to the center of the image.

What do you think?

**Comment by Santhosh-Sellavel** [comment]
@bernhardoj It's fine to have a focus outline (blue line) on the image when the user intends to have so. But no on the case while pressing ESC.

**Comment by Santhosh-Sellavel** [comment]
@bernhardoj This looks like a separate issue to me. So it's better you start with reporting on #expensify-bugs

> Btw, here is the error when we try to zoom in with enter key:
> <img alt="image" width="588" src="https://user-images.githubusercontent.com/50919443/234737395-e89f41d0-7446-4b7d-89fb-cccc196cc1df.png">

cc: @Beamanator

**Comment by bernhardoj** [comment]
That's a normal behavior of web to have the blue border on pressing any key when an element is focused. (even this closed issue https://github.com/Expensify/App/issues/10599 mentioned the Esc key)

Based on @Beamanator comment here https://github.com/Expensify/App/issues/17419#issuecomment-1517609195, we want to handle this because the detail image is not an input. I assume the "input" here means an interactable element. I also thought that we can't interact with the image by keyboard, thus I offer to completely disallow the image to receive a focus from either mouse or keyboard. But I was wrong which lead to my last [comment](https://github.com/Expensify/App/issues/17419#issuecomment-1529587841).

Someone already took the report here https://expensify.slack.com/archives/C049HHMV9SM/p1682691050575909, but marked as not in scope.

**Comment by jjcoffee** [comment]
@bernhardoj @situchan I'm not sure this is quite the same as #10599 as that issue was closed as the focusing there (onto the avatar _after_ closing) is correct from an accessibility perspective, whereas here the focus is flashing onto the modal just before it closes, which is clearly not so useful.

**Comment by Beamanator** [comment]
Ok yes so we agreed this issue is not the same as https://github.com/Expensify/App/issues/10599 👍 

> It's fine to have a focus outline (blue line) on the image when the user intends to have so. But no on the case while pressing ESC.

Agreed 👍 

> @Beamanator in case you missed my https://github.com/Expensify/App/issues/17419#issuecomment-1524396931, I think it's fine to have the blue border because we can interact (zoom in/out) with the avatar by keyboard.

@bernhardoj Interesting point here! Do you think that's basically what's happening when we click (with mouse) to zoom on the picture as well? 

Also I only see the blue border when clicking ESC after zooming in / out on the Concierge logo, does anyone see the blue border on other avatars?

**Comment by jjcoffee** [comment]
@Beamanator The blue border appears if you have any PNG with transparency at the edges (which Concierge has). Normal images are "hiding" the focus outline (if you go into DevTools and reduce their width % a little, you'll see the same behaviour for all images).

**Comment by jjcoffee** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
We want to stop the avatar image from being focused just as the modal is closed as this serves no accessibility purpose.

### What is the root cause of that problem?
When you click the image the first time, it gains focus. When you then press the `Escape` key (or in fact any key), this triggers `focus-visible` on the image element.

### What changes do you think we should make in order to solve the problem?

For the `AttachmentCarousel` we are currently blurring on every `keyPress`:

https://github.com/Expensify/App/blob/081e45ce5de0fef33edcd7b909d4e4f9f6877efb/src/components/AttachmentCarousel/CarouselActions/index.js#L29-L38

Which means there is no highlighting going on at all and the only supported keyboard navigation is left/right arrow presses.

Unless we think we should support extra keyboard navigation to zoom in/out there, I think we should also not support it here for consistency. Therefore we should implement a similar `handleKeyPress` function in `Modal`, by adding a `keyup` listener to the document in `componentDidMount` which calls `handleKeyPress`:

```
handleKeyPress(event) {
	event.target.blur();
}
```

### What alternative solutions did you explore? (Optional)
If we want to still support other keyboard actions, the same can be done as above but with a conditional to only blur on Escape key press:

```
handleKeyPress(event) {
	if (event.key === 'Escape') {
		event.target.blur();
	}
}
```

We'd still be left with the weird behaviour that any other key will also cause the image to get highlighted (but maybe this isn't such a strong concern).

**Comment by MelvinBot** [comment]
Looks like something related to `react-navigation` may have been mentioned in this issue discussion.

As a reminder, please make sure that all proposals are not workarounds and that any and all attempt to fix the issue holistically have been made before proceeding with a solution. Proposals to change our `DeprecatedCustomActions.js` files should not be accepted.

Feel free to drop a note in [#expensify-open-source](https://expensify.slack.com/archives/C01GTK53T8Q) with any questions.

**Comment by bernhardoj** [comment]
1. My gut telling me it isn't right 🤣
2. The only solution I could think of is to blur the active element on pressing the ESC.
3. If we are still going on to solve this, we need to do the same with the X button because this "issue" happens to every interactable elements.

> Or do you think the blue border appearing is actually useful somehow?

I agree it's not useful when we are closing the modal, but I also think it doesn't add any value to prevent the blue border showing.

But it's all back to your decision. I'm not an expert on accessibility and just sharing my thoughts 😄.

**Comment by Santhosh-Sellavel** [comment]
For everyone, we are missing something else here. This happens in a few other places also here is one more https://github.com/Expensify/App/issues/18052

The solution should not be any workaround. We would need a clear RC and solution

> @anmurali Looks like we should double the price of this one, what do you think @Santhosh-Sellavel?

@Beamanator Yeah let's double this one!

**Comment by MelvinBot** [comment]
📣 It's been a week! Do we have any satisfactory proposals yet? Do we need to adjust the bounty for this issue? 💸

**Comment by Beamanator** [comment]
So coming back to this (sorry for the delay) I think we need to determine if we want anything to highlight AT ALL after clicking "Escape", here's why: 

Someone suggested that it's pretty common to highlight the last active DOM element when pressing "Escape". I tested this flow:
1. Click on a chat report comment
2. Move the mouse away
3. Press "Escape"
4. Notice that the chat you clicked on has the "Blue focused" box around it, and you can move it around by pressing Tab

This is _interesting_ but do we want it? I Haven't seen this behavior in any other app (like whatsapp, slack, gmail) so is it desired or not? Maybe it's an accessibility thing? I don't know. So I think we should bring this to slack, @Santhosh-Sellavel if you agree do you mind bringing up this topic in slack please?

**Comment by MelvinBot** [comment]
@Beamanator @anmurali @Santhosh-Sellavel this issue is now 3 weeks old. There is one more week left before this issue <a href='https://stackoverflowteams.com/c/expensify/questions/15233/15234#15234'>breaks WAQ</a> and will need to go internal. What needs to happen to get a PR in review this week? Please create a thread in #expensify-open-source to discuss. Thanks!

**Comment by Santhosh-Sellavel** [comment]
Sorry for the delay @Beamanator 
I tried the same steps as said [here](https://github.com/Expensify/App/issues/17419#issuecomment-1534424666). As far as I tested It's working on Slack, GitHub, and Gmail as well.

https://user-images.githubusercontent.com/85645967/236690366-d368d64d-7528-49be-b0f5-ef7ec6af8989.mov

But with or without pressing `ESC`, on pressing `Tab` the Tab navigation is enabled from the place where the user last interacted on the page.

But on pressing `ESC` nothing got focused, only pressing `Tab` initiated the Tab selection/navigation which is an accessibility thing.

**Comment by Beamanator** [comment]
Aah interesting @Santhosh-Sellavel - great investigation! So hmm I like that pressing `Tab` does start navigating you around the page, from where the user last interacted on the page. But it's interesting that nothing ever gets highlighted directly when pressing `ESC` 🤔 

Anyway, it's time we bring this up in Slack

**Comment by Beamanator** [comment]
We got enough feedback saying let's do nothing, so closing this out 👍
